Fail to build coq 8.7.1 with ocaml 4.06

I am trying to update the Arch Linux package against ocaml 4.06 https://www.archlinux.org/packages/community/x86_64/coq/

I already fixed the issue reported in #6513 (by applying 8e7b876b6d9f3d4131e3b7e969f8e1943154df4c and packaging `ocaml-num`)

However there is still a build issue further on:

```
COQMKTOP -o bin/coqtop
findlib: [WARNING] Interface profile.cmi occurs in several directories: /usr/lib/ocaml/compiler-libs, lib
findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /usr/lib/ocaml, /usr/lib/ocaml/compiler-libs
File "/tmp/coqmain9aadf4.ml", line 1:
Error: Cannot find file nums.cmxa
make[1]: *** [Makefile.build:391: bin/coqtop] Error 2
make[1]: Leaving directory '/build/coq/src/coq-8.7.1'
make: *** [Makefile:159: submake] Error 2
```

`nums.cmxa` exists in `/usr/lib/ocaml/num/` thanks to https://www.archlinux.org/packages/community/x86_64/ocaml-num/ (see the file list).  For this `ocaml-num` package I used the findlib installation method from this commit https://github.com/ocaml/num/commit/7dd5e935aaa2b902585b3b2d0e55ad9b2442fff0

Full build log: https://paste.aliens-lyon.fr/kcX

Any idea?  I looked at other distributions but it seems that nobody is packaging ocaml 4.06 yet.

--- a/Makefile.build
+++ b/Makefile.build
@@ -530,7 +530,7 @@ CSDPCERTCMO:=lib/clib.cma $(addprefix plugins/micromega/, \
 
 $(CSDPCERT): $(call bestobj, $(CSDPCERTCMO))
 	$(SHOW)'OCAMLBEST -o $@'
-	$(HIDE)$(call bestocaml,,nums unix)
+	$(HIDE)$(call bestocaml,-linkpkg -package num -package unix,)
 
 ###########################################################################
 # tests

--- a/tools/coqmktop.ml
+++ b/tools/coqmktop.ml
@@ -108,7 +108,7 @@ let incl_all_subdirs dir opts =
 
 (** OCaml + CamlpX libraries *)
 
-let ocaml_libs = ["str.cma";"unix.cma";"nums.cma";"dynlink.cma";"threads.cma"]
+let ocaml_libs = ["str.cma";"dynlink.cma"]
 let camlp4_libs = ["gramlib.cma"]
 let libobjs = ocaml_libs @ camlp4_libs
 
@@ -289,6 +289,7 @@ let main () =
       List.filter ((<>) "") (split_on_char ' ' Coq_config.caml_flags) in
     let args =
       coq_camlflags @ "-linkall" :: "-w" :: "-31" :: flags @ copts @ options @
+      ["-linkpkg"; "-package"; "num"] @
       (std_includes basedir) @ tolink @ [ main_file ] @ topstart
     in
     if !echo then begin

